src/components/TestRunList.tsx

```diff
@@ -56,8 +59,61 @@ const TestRunList: React.FunctionComponent<{
     setDeleteDialogOpen(!deleteDialogOpen);
   };
 
+  const doesHaveUnapprovedImage = (): boolean => {
+    let status = false;
+    items.forEach((eachItem) => {
+      if (eachItem.status === TestStatus.new || eachItem.status === TestStatus.unresolved) {
+        status = true;
+        return;
+      }
+    });
+    return status;
+  };
+
+  const approveAll = () => {
+    enqueueSnackbar("Wait for the confirmation message when approval is completed.", {
+      variant: "info",
+    });
+    let lastItemToModify: TestRun | null = null;
+    //Determine the last item to be modified.
+    items.forEach((eachItem) => {
+      if (!(eachItem.status === TestStatus.ok || eachItem.status === TestStatus.approved)) {
+        lastItemToModify = eachItem;
+      }
+    });
+    //Iterate through all unapproved and send message when last item to modify is encounterd.
+    items.forEach((eachItem) => {
+      if (!(eachItem.status === TestStatus.ok || eachItem.status === TestStatus.approved)) {
+        testRunService
+          .approve(eachItem.id, false)
+          .then((testRun) => {
+            updateTestRun(testRunDispatch, testRun);
+            if (eachItem === lastItemToModify) {
+              enqueueSnackbar("All approved", {
+                variant: "success",
+              });
+            }
+          })
+          .catch((err) =>
+            enqueueSnackbar(err, {
+              variant: "error",
+            })
+          );
+      }
+    });
+  };
+
   return (
     <React.Fragment>
+      {items.length !== 0 && doesHaveUnapprovedImage() && <Button
```

Comment:
`items.length !== 0` should not make any difference on this condition as `doesHaveUnapprovedImage` covers this case